### PR TITLE
NETOBSERV-736: Added downstream deployment configuration for monitoring collection

### PR DESCRIPTION
**Pull request by OlivierCazade**
Add downstream configuration boolean.

To override this boolean run: `make set-release-kind-downstream`

This PR does not change the HTTPS config for the operator metrics endpoint but I did not want to overload the PR, I will do another PR about it.

### CODE REVIEW

File: bundle/manifests/netobserv-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -649,6 +653,8 @@ spec:
                   value: quay.io/netobserv/flowlogs-pipeline:v0.1.8
                 - name: RELATED_IMAGE_CONSOLE_PLUGIN
                   value: quay.io/netobserv/network-observability-console-plugin:v0.1.9
+                - name: DOWNSTREAM_DEPLOYMENT
```
Comment (memodi) [inline bundle/manifests/netobserv-operator.clusterserviceversion.yaml:656]: would `DOWNSTREAM_DEPLOYMENT ` be set to true in CPAAS pipeline stages?

File: main.go [hunk 2]
```diff
@@ -91,6 +91,7 @@ func main() {
 	flag.StringVar(&config.EBPFAgentImage, "ebpf-agent-image", "quay.io/netobserv/netobserv-ebpf-agent:main", "The image of the eBPF agent")
 	flag.StringVar(&config.FlowlogsPipelineImage, "flowlogs-pipeline-image", "quay.io/netobserv/flowlogs-pipeline:main", "The image of Flowlogs Pipeline")
 	flag.StringVar(&config.ConsolePluginImage, "console-plugin-image", "quay.io/netobserv/network-observability-console-plugin:main", "The image of the Console Plugin")
+	flag.BoolVar(&config.DownstreamDeployment, "downstream-deployment", false, "Either this deployment is a downstream deployment ot not")
```
Comment (memodi) [inline main.go:94]:
```suggestion
	flag.BoolVar(&config.DownstreamDeployment, "downstream-deployment", false, "Either this deployment is a downstream deployment or not")
```

**Review by memodi — CHANGES_REQUESTED** [review]

File: bundle/manifests/netobserv-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (OlivierCazade) [reply bundle/manifests/netobserv-operator.clusterserviceversion.yaml:656]: Yes, the last step for this task is to update downstream build to change DOWNSTREAM_DEPLOYMENT
We already do this kind of changes for container images such as RELATED_IMAGE_CONSOLE_PLUGIN

Comment (memodi) [reply bundle/manifests/netobserv-operator.clusterserviceversion.yaml:656]: got it, thanks!. Could you open a PR for downstream setting this env to true? 

we'd have to test this PR post-merge cc @jotak

**Comment by memodi** [comment]
@OlivierCazade - I am not sure I am testing this the right way:
1. I modified manager.yaml to set `DOWNSTREAM_DEPLOYMENT == true`
2. deployed image from this PR
3. applied serviceMonitor as [described](https://raw.githubusercontent.com/netobserv/network-observability-operator/55bef5bd0be291eff5d9d6563cc707c38ccd69a2/config/prometheus/monitor.yaml) in PR #273 in NS `netobserv`
```
$ oc get servicemonitor
NAME              AGE
metrics-monitor   25m
```
but I am not able to see metrics for NOO when I go under `Observe --> Metrics` on Console, could you let me know what am I missing in above procedure? Also, this PR needs a rebase.

**Comment by memodi** [comment]
thanks @OlivierCazade , this info helped:

> PR #282 is about tagging the namespace so the platform monitoring stack collect the metrics and not the user workload monitoring stack.
> If you want operator metrics, yes you will need both PR, but what you should see by testing 282 alone is the metrics from FLP getting collected automatically without enabling user workload monitoring.

I am able to see FLP metrics based off changes in this PR, modified `DOWNSTREAM_DEPLOYMENT` to set to true in `manager.yaml` , will re-test for downstream build

/label qe-approved

**Review by jotak — APPROVED** [review]
lgtm (not tested)

**Pull request merged**